### DISCUSSION TITLE
[Q] How to provide user/client with an option to download an excel file .

### DISCUSSION THREAD

**Discussion by Srezzx** [exceljs/exceljs#2447] [Q&A]
The code I am using is - 
var workbook = new Excel.Workbook();	
	workbook.xlsx.readFile('sheet1.xlsx').then(function(){
		var worksheet = workbook.getWorksheet(1);
		var row = worksheet.getRow(5);
		row.getCell(1).value = 188;
		row.commit();
		return workbook.xlsx.writeFile('sheet1.xlsx');
	});

i tried adding
- res.attachment("sheet1.xlsx");
but it is downloading a corrupted file 

How can I make the sheet1.xlsx available for download to the user?
Please help 
Thank you for the co-operation

**Comment by trigg-mosaic** [comment]
Check here:  [How to use ExcelJs in the ClientSide](https://github.com/exceljs/exceljs/issues/768#issuecomment-503882112)

This method worked for me. In mine, I changed the example to use a promise instead of await, though:

```
wb.xlsx.writeBuffer()
        .then((buf) => {
            saveAs(new Blob([buf]), fileName);
        });
```

**Comment by OmkarKumbare** [comment]
Used exceljs for implementation

Here is simple example for export excel file using react [StackBlitz](https://stackblitz.com/edit/react-export-to-excel?file=src%2FApp.js)
